src/app/clusters/soil-measurement-server/tests/TestSoilMeasurementCluster.cpp

```diff
@@ -66,83 +61,94 @@ class SoilMeasurementClusterLocal : public SoilMeasurementCluster
     SoilMoistureMeasuredValue::TypeInfo::Type GetSoilMoistureMeasuredValue() const { return mSoilMoistureMeasuredValue; }
 };
 
-class SoilMeasurementClusterTest
+struct TestSoilMeasurementCluster : public ::testing::Test
 {
-    SoilMeasurementClusterLocal soilMeasurement;
+    static void SetUpTestSuite() { ASSERT_EQ(chip::Platform::MemoryInit(), CHIP_NO_ERROR); }
 
-public:
-    template <typename... Args>
-    SoilMeasurementClusterTest(Args &&... args) : soilMeasurement(std::forward<Args>(args)...)
+    static void TearDownTestSuite() { chip::Platform::MemoryShutdown(); }
+
+    void SetUp() override { ASSERT_EQ(soilMeasurement.Startup(context), CHIP_NO_ERROR); }
+
+    void TearDown() override { soilMeasurement.Shutdown(); }
+
+    TestSoilMeasurementCluster() :
+        testContext(), context(testContext.Create()),
```

Comment:
```suggestion
        context(testContext.Create()),
```

does this work? Seems simpler.